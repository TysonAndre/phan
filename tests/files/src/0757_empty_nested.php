<?php
function test757(array $additional) {
    if (empty($additional['product']['comment'])) {
        // Should not warn
        echo $additional['product']['listprice'] . "\n";
    }
    if (empty($additional['product'])) {

        // should warn
        echo $additional['product']['listprice'] . "\n";
    }
}
/**
 * @param int $x
 */
function test757b($x) {
    if (empty($x)) {
        if (is_object($x)) {
            echo "Impossible\n";
        } elseif (is_array($x)) {
            echo "The empty array\n";
        } elseif (is_resource($x)) {
            echo "Impossible\n";
        }
    }
}
