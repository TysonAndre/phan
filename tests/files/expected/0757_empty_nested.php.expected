%s:10 PhanTypeArraySuspiciousNull Suspicious array access to null
%s:10 PhanTypeSuspiciousStringExpression Suspicious type null of a variable or expression $additional['product']['listprice'] used to build a string. (Expected type to be able to cast to a string)
%s:18 PhanImpossibleCondition Impossible attempt to cast $x of type ?''|?'0'|?0|?0.0|?array{}|?false to object
%s:22 PhanImpossibleCondition Impossible attempt to cast $x of type ?''|?'0'|?0|?0.0|?false to resource
